### DISCUSSION TITLE
decorated, un-assigned no longer throws error (and breaks action.bound)

### DISCUSSION THREAD

**Discussion by alisman** [mobxjs/mobx#2655] [General]
In trying to upgrade Mobx 4 to 6, we just spent a few hours debugging a situation where we han observable decorator on an un-assigned class property.    This curiously caused action.bound decorator to fail on an unrelated method!    I understand we are kind of asking for trouble by continuing to use the now marginalized decorators (true/false?) but our code base is so dependent on them and we also really like them.   But anyhow, i think there used to be an error message when you left an observable undefined?   Anyway we can catch this so we don't run into it again?

```
class something {
 
    // leaving this un-assigned is illegal and strangely breaks the action.bound below
    // must be something with the way decorators are now handled
    @observable someVar;

    @action.bound doSomething(){
           // this is not bound due to the un-assigned property
    }

}

```

**Comment by mweststrate** [comment]
Did you update your compiler settings according to the migration guide? Can
you provide a minimal repro?

**Reply by alisman** [reply]
We will double check compiler settings, thanks!  @mweststrate just wanted to say a huge thank you.  it is INCREDIBLE that you respond to these kind of messages.  i hope that FB is paying you very well b/c you deserve to be rewarded for your contribution to web development.   Mobx is at the heart of our application at cbioportal.org, a data visualization platform that is used by cancer researchers all around the world.  So, kudos to you.
